Sources/Core/Shared/Operation.swift

```diff
@@ -76,6 +76,24 @@ public class Operation: NSOperation {
         }
     }
 
+    /** 
+     Type to express the intent of the user in regards to executing an Operation instance
+     
+     - see: https://developer.apple.com/library/ios/documentation/Performance/Conceptual/EnergyGuide-iOS/PrioritizeWorkWithQoS.html#//apple_ref/doc/uid/TP40015243-CH39
+    */
+    public enum UserIntent: Int {
+        case Background = 0, SideEffect, Initiated
+
+        internal var qos: NSQualityOfService {
+            switch self {
+            case .Initiated, .SideEffect:
+                return .UserInitiated
+            default:
+                return .Default
+            }
```

Comment:
The `.SideEffect` case is to describe operations which are triggered as a side effect of user actions. So in your Post example, after the `DownloadPostOperation` finishes, perhaps a parent groups goes off and downloads all the Comment items - `DownloadCommentOperation` - you might consider these to be side effects.

My thinking was that semantically, they might have different cancellation, early termination or error handling.